### PR TITLE
Do not permit function names containing registers ##anal

### PR DESCRIPTION
**Pull request by radare**

### CODE REVIEW

**Comment by radare** [comment]
There are at least 3 wrong tests: https://api.travis-ci.org/v3/job/495272289/log.txt

screenshot 2019-02-19 at 06 35 49

File: libr/core/cmd_anal.c [hunk 1]
```diff
@@ -2181,50 +2181,86 @@ static bool fcnNeedsPrefix(const char *name) {
...
+	fcn->name = name;
+	if (core->anal->cb.on_fcn_rename) {
+		core->anal->cb.on_fcn_rename (core->anal,
+				core->anal->user, fcn, name);
 	}
-	char *oname = fcn->name;
+	return true;
+}
+
+#if 0
```
Comment (ret2libc) [inline libr/core/cmd_anal.c]: remove all this thing :/ there is git if you want to have a look at old stuff, it just makes review so much harder

**Comment by radare** [comment]
Thats the only command i dont need because its obvious. Can you review the rest?

**Comment by ret2libc** [comment]
I know, but it stil makes the diff much harder to read than if you just remove it.

Anyway, I don't feel familiar enough with this part of the code to really understand what's going on, sorry :(

**Pull request merged**